packages/lib-classifier/src/store/SubjectStore.spec.js

```diff
@@ -51,97 +46,78 @@ describe('Model > SubjectStore', function () {
-          rootStore.subjects.advance()
-        }
+        rootStore = setupStores()
       })
 
       after(function () {
         clientStub.panoptes.get.restore()
+        rootStore = null
       })
 
       it('should leave the active subject empty', function () {
+        while (rootStore.subjects.resources.size > 0) {
+          rootStore.subjects.advance()
+        }
         expect(rootStore.subjects.resources.size).to.equal(0)
         expect(rootStore.subjects.active).to.be.undefined()
       })
     })
   })
 
-  describe('Views > isThereMetadata', function (done) {
-    it('should return false when there is not an active queue subject', function () {
-      rootStore = RootStore.create(
-        { projects: ProjectStore.create(), subjects: SubjectStore.create(), workflows: WorkflowStore.create() },
-        { client: {
-          panoptes: {
-            get: () => {
-              return Promise.resolve({
-                body: { subjects: [] }
-              })
-            }
-          }
-        }
-        }
-      )
+  describe('Views > isThereMetadata', function () {
+    let rootStore
+    afterEach(function () {
+      rootStore = null
+    })
+
+    it('should return false when there is not an active queue subject', function (done) {
+      const getStub = stubPanoptesJs({ subjects: [] })
 
-      setupStores(rootStore)
+      rootStore = setupStores(getStub)
       rootStore.subjects.populateQueue().then(() => {
         expect(rootStore.subjects.isThereMetadata).to.be.false()
       }).then(done, done)
     })
 
     it('should return false if the active subject does not have metadata', function (done) {
-      rootStore = RootStore.create(
-        { projects: ProjectStore.create(), subjects: SubjectStore.create(), workflows: WorkflowStore.create() },
-        { client: clientStub }
-      )
-
-      setupStores(rootStore)
+      rootStore = setupStores()
       rootStore.subjects.populateQueue().then(() => {
-        expect(Object.keys(rootStore.subjects.active.toJSON().metadata)).to.have.lengthOf(0)
+        expect(Object.keys(rootStore.subjects.active.metadata)).to.have.lengthOf(0)
         expect(rootStore.subjects.isThereMetadata).to.be.false()
       }).then(done, done)
     })
 
     it('should return false if the active subject only has hidden metadata', function (done) {
       const subjectWithHiddenMetadata = SubjectFactory.build({ metadata: { '#foo': 'bar' } })
-      rootStore = RootStore.create(
-        { projects: ProjectStore.create(), subjects: SubjectStore.create(), workflows: WorkflowStore.create() },
-        {
-          client: {
-            panoptes: {
-              get: () => {
-                return Promise.resolve({
-                  body: { subjects: [subjectWithHiddenMetadata] }
-                })
-              }
-            }
-          }
-        }
-      )
+      const getStub = stubPanoptesJs({ subjects: subjectWithHiddenMetadata })
```

Comment:
Should `subjects` be an array here?

Reply:
`stubPanoptesJs` can take either a single resource stub or an array of resource stubs. A single stub will be put into an array.